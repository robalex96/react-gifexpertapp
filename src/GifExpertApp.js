import React, { Fragment, useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({categoriesDefault = []}) => {
    const [categories, setCategories] = useState(categoriesDefault);

    return (
        <Fragment>
            <h2>Gif Expert App</h2>
            <AddCategory setCategories={setCategories}/>
            <hr />
            {
                categories.map((category) =>  
                    <GifGrid
                        key={category}
                        category={category}
                    />
                )
            }
        </Fragment>
    )
}