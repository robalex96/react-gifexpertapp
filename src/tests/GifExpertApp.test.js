import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('GifExpertApp', () => {
    test('Render correctly ', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Render two categories', () => {
        const categoriesDefault = ['One Punch', 'Goku'];        
        const wrapper = shallow(<GifExpertApp categoriesDefault={categoriesDefault} />);
        expect(wrapper.find('GifGrid').length).toBe(2);
    });
    
});