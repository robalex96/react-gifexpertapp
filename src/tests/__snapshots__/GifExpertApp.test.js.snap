// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Render correctly  1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
